### PR TITLE
Refactor request client chain responsibilities

### PR DESCRIPTION
**Pull request by armstnp**

### CODE REVIEW

File: lib/smartsheet/api/request.rb [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        @endpoint_spec = endpoint_spec
+        @request_spec = request_spec
+      end
+
+      def method
+        endpoint_spec.method
+      end
+
+      def url
+        Smartsheet::API::UrlBuilder.new(endpoint_spec, request_spec).build
```
Comment (cameronbowie) [inline lib/smartsheet/api/request.rb]: Could you pull this into the initialize? Otherwise, we are building the URL each time it is requested

File: lib/smartsheet/api/retrying_net_client_adapter.rb [hunk 2]
```diff
@@ -1,16 +1,16 @@
 module Smartsheet
   module API
-    class RetryingNetClient
+    class RetryingNetClientAdapter
       RETRY_CHECK = ->(response) { response.should_retry? }
 
-      def initialize(client, retrier = nil)
+      def initialize(client, retrier = RetryLogic.new)
```
Comment (cameronbowie) [inline lib/smartsheet/api/retrying_net_client_adapter.rb]: Could you remove this default arg?

File: lib/smartsheet/api/retrying_net_client_adapter.rb [hunk 3]
```diff
@@ -1,16 +1,16 @@
 module Smartsheet
   module API
-    class RetryingNetClient
+    class RetryingNetClientAdapter
```
Comment (cameronbowie) [inline lib/smartsheet/api/retrying_net_client_adapter.rb]: Why did you rename this? This seems to be more a net client than an adapter. It adheres to the same interface as the other net clients

Comment (armstnp) [reply lib/smartsheet/api/retrying_net_client_adapter.rb]: Adapter -> Decorator

Comment (armstnp) [reply lib/smartsheet/api/retrying_net_client_adapter.rb]: Retrying->Retry

File: lib/smartsheet/api/retrying_net_client_adapter.rb (hunk 2 above)
Comment (armstnp) [reply lib/smartsheet/api/retrying_net_client_adapter.rb]: :+1:

File: test/smartsheet/api/faraday_net_client_test.rb [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    faraday_request.expects(:'headers=').with('headers')
+    faraday_request.expects(:'params=').with('params')
+    faraday_request.expects(:'body=').with('body')
+
+    @response = mock
+    @response.stubs(:body).returns Smartsheet::API::Response.from_result({})
+
+    conn = mock
+    conn.stubs(:get).yields(faraday_request).returns @response
+    conn.expects(:use).with(Smartsheet::API::Middleware::ErrorTranslator)
```
Comment (cameronbowie) [inline test/smartsheet/api/faraday_net_client_test.rb]: For clarity, it might make sense to move the mocks to a helper method that is called with each test (rather than in the 'before'
